app/models/rent.rb

```diff
@@ -0,0 +1,17 @@
+class Rent < ApplicationRecord
+  belongs_to :book
+  belongs_to :user
+
+  validates :start_date, :end_date, presence: true
+  validate :end_date_after_start_date
+
+  private
+
+  def end_date_after_start_date
+    return if end_date.blank? || start_date.blank?
+
+    return unless end_date <= start_date
```

Comment:
Intenta siempre priorizar `if` que `unless` (tiene una mejor lectura)

```suggestion
    return if end_date > start_date
```